### DISCUSSION TITLE
Differences between time-dependent and steady state simulation

### DISCUSSION THREAD

**Discussion by ZhixinFan-hub** [svenreiche/Genesis-1.3-Version4#117] [Q&A] [answered]
Hi
I am testing the differences between steady-state and time-dependent simulation.
I keep every parameter the same and just switch "time=true" & "time=false".
There is no field output in the time-dependent mode but for the steady-state mode, everything goes well.
I don't know why.
Attached please find my files and results.
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/65019083/f6f8b157-0f6d-4d36-9ad7-da593b26b686)
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/65019083/a1ffaa1e-98a4-4604-a67c-24ebcdeac15b)
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/65019083/9e0f7159-67c1-4ffa-b9ee-e7d9a2094cb7)
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/65019083/e5b19c99-ac66-4130-b912-2d9842e6919c)

**Comment by ZeugAusHH** [comment]
Hi,

one quite obvious issue with your simulation input file is the value of `slen` in the `&time` block. Its unit is in meters, so for 95 micron, you have to specify 95E-6. Actually interesting that data was written in this case...

In my opinion, input files for steady-state simulations should not have a `&time` block at all, see for instance: https://github.com/svenreiche/Genesis-1.3-Version4/tree/master/examples/Example1-SteadyState .

By the way: If you run a recent version of "GENESIS 1.3", you can specify in your `&setup` block: "write_semaphore_file=true". Then, if the simulation run was successful, a zero-byte file with extension `.sema` is generated.

Hope this helps.

Best regards from Hamburg

**Reply by ZhixinFan-hub** [reply]
Hi
Honestly, even though I adjusted the "slen", the results are still nothing.
In time-dependent mode, the field values are always 0, I don't know why.
Also, in steady state simulation, I set "time=false".

**Reply by ZeugAusHH** [reply]
Hi,

could you then please provide your input and lattice files, as well as the terminal output (stdout and stderr) of your time-dependent simulation run? Please be sure to attach the actual files, *not screenshots*. Then people can have a look.

Concerning the steady-state simulations, I would completely leave out the `&time` block.

Best regards

**Reply by ZhixinFan-hub** [reply]
Input file:
```
&setup
rootname=prxt
lattice=prxt.lat
beamline=FEL
lambda0=1e-10
gamma0=1135
delz=0.1
seed=1
shotnoise=true
one4one=true
write_semaphore_file=true
&end

&lattice
zmatch=9.5
&end

&field
power=5e3
dgrid=2.000000e-04
ngrid=255
waist_size=30e-6
&end

&beam
current=3000
delgam=1.000000
ex=4.000000e-07
ey=4.000000e-07
&end

&time
s0=0
slen=0.1
sample=1
time=true
&end

&track
output_step=1
field_dump_step=1
beam_dump_step=1
field_dump_at_undexit=true
sort_step=1
&end

&write
field=fld
beam=beam
&end
```
Lattice file:
```
D1: DRIFT = { l = 0.44};
D2: DRIFT = { l = 0.24};
QF: QUADRUPOLE = { l = 0.080000, k1= 2.000000 };
QD: QUADRUPOLE = { l = 0.080000, k1= -2.000000 };
UND: UNDULATOR = { lambdau=0.015000, nwig=266, aw=0.84853, helical= True};

FODO: LINE={UND,D1,QF,D2,UND,D1,QD,D2};

FEL: LINE={2*FODO};
```

**Reply by ZeugAusHH** [reply, marked as answer]
Hi,

many thanks for the files, I had a look. The main issue may be that you specify `&time` too late, when the field is generated you already have to know the number of slices...

Please find my modified `.in` file in the attachment (I set zstop for debugging). Please note that I added a zero to the gamma0 to match the resonance condition.

Best regards

[prxt.in.txt](https://github.com/svenreiche/Genesis-1.3-Version4/files/13126763/prxt.in.txt)

**Reply by ZeugAusHH** [reply]
Update: Filed new issue https://github.com/svenreiche/Genesis-1.3-Version4/issues/118 .
